app/controllers/minor/routes.py

```diff
@@ -1,15 +1,188 @@
-from flask import g, render_template, request, abort, flash, redirect, url_for
+from flask import g, render_template, request, abort, flash, redirect, url_for, jsonify
 from peewee import DoesNotExist
 
 from app.controllers.minor import minor_bp
 from app.models.user import User
 from app.models.term import Term
-from app.models.attachmentUpload import AttachmentUpload
+from app.models.summerExperience import SummerExperience
+from app.models.otherExperience import OtherExperience
 
 from app.logic.fileHandler import FileHandler
 from app.logic.utils import selectSurroundingTerms, getFilesFromRequest
 from app.logic.minor import getProgramEngagementHistory, getCourseInformation, getCommunityEngagementByTerm, removeSummerExperience
-from app.logic.minor import saveOtherEngagementRequest, setCommunityEngagementForUser, saveSummerExperience, getSummerTerms, getSummerExperience, getEngagementTotal
+from app.logic.minor import saveOtherEngagementRequest, setCommunityEngagementForUser, saveSummerExperience, getSummerTerms, getSummerExperience, getEngagementTotal, createSummerExperience, createOtherEngagement
+import logging
+
+
+# ################################################## SUMMER EXPERIENCE START ###########################################################
+
+@minor_bp.route('/cceMinor/<username>/addSummerExperience', methods=['POST'])
+def addASummerExperience(username):
+    try:
+        createSummerExperience(username, request.form)
+        flash(f'Summer Experience added successfully by {username}', 'success')
+        return redirect(url_for('minor.viewProposal', username=username, activeTab='trainingEvents'))
+    except Exception as e:
+        flash(f'An error occurred while adding the summer experience: {e}', 'danger')
+        logging.error(f'An error occurred while adding the summer experience: {e}')
+    return redirect(url_for('minor.viewCceMinor', username=username))
+
+
+
+@minor_bp.route('/cceMinor/<username>/viewProposal', methods=['GET'])
+def viewProposal(username):
```

Comment:
This route duplicates the already existing page.

Reply:
resolved